### PR TITLE
SDIT-1125 Refactor tests ready for change to role access

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingCaseNotesResourceIntTest.kt [hunk 1]
```diff
@@ -563,6 +563,4 @@ class BookingCaseNotesResourceIntTest : ResourceTest() {
     val ocn = offenderCaseNoteRepository.findById(caseNoteId).get()
     offenderCaseNoteRepository.delete(ocn)
   }
-
-  internal fun String.readFile(): String = this@BookingCaseNotesResourceIntTest::class.java.getResource(this).readText()
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingCaseNotesResourceIntTest.kt:567]: unused

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_getHousingLocation.kt [hunk 2]
```diff
@@ -50,6 +55,9 @@ class OffenderResourceIntTest_getHousingLocation : ResourceTest() {
       .responseBody.blockFirst()!!
 
     assertThat(location).isEqualTo(OffenderLocation())
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_getHousingLocation.kt:57]: This test was interferring with the refactored Kotlin (from Java) test so needed tidying up code adding

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderSentenceResourceImplIntTest.kt [hunk 3]
```diff
@@ -0,0 +1,711 @@
...
+        .uri("/api/offender-sentences/booking/-2/home-detention-curfews/latest/checks-passed")
+        .headers(setClientAuthorisation(listOf("ROLE_VIEW_PRISONER_DATA")))
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isForbidden
+    }
+  }
+
+  @Nested
+  @DisplayName("PUT /api/offender-sentences/booking/{bookingId}/home-detention-curfews/latest/approval-status")
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderSentenceResourceImplIntTest.kt:584]: I just noticed there are a few cucumber tests relating to the curfew endpoints - and these are the ones that I am changing roles for.   So will do as separate PR

**Review by steverendell — APPROVED** [review]
lgtm
